cli/napi/js_native_api.rs

```diff
@@ -154,7 +259,8 @@ fn napi_create_buffer_copy(
   result_data: *mut *mut u8,
   result: *mut napi_value,
 ) -> Result {
-  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  check_env!(env);
+  let env = unsafe { &mut *env };
```

Comment:
Why this instead of just `let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;`

Reply:
Because `.as_mut()` consumes the pointer and makes it impossible to create a value out of scope, assign it to the result pointer and later call a function like `napi_clear_error_code`